x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/ExactMatchProcessorTests.java

```diff
@@ -166,6 +167,43 @@ public void testIgnoreKeyMissing() throws Exception {
         }
     }
 
+    public void testExistingFieldWithOverrideDisabled() throws Exception {
+        MockSearchFunction mockSearch = mockedSearchFunction(Map.of("elastic.co", Map.of("globalRank", 451, "tldRank",23, "tld", "co")));
+        ExactMatchProcessor processor = new ExactMatchProcessor("_tag", mockSearch, "_name", "domain", false, false,
+            List.of(new EnrichSpecification("tld", "tld")));
+
+        IngestDocument ingestDocument = new IngestDocument(new HashMap<>(Map.of("domain", "elastic.co", "tld", "tld")), Map.of());
+        IngestDocument[] resultHolder = new IngestDocument[1];
+        Exception[] exceptionHolder = new Exception[1];
+        processor.execute(ingestDocument, (result, e) -> {
+            resultHolder[0] = result;
+            exceptionHolder[0] = e;
+        });
+        assertThat(exceptionHolder[0], nullValue());
+        assertThat(resultHolder[0].hasField("tld"), equalTo(true));
+        assertThat(resultHolder[0].getFieldValue("tld", Object.class), equalTo("tld"));
+    }
+
+    public void testExistingNullFieldWithOverrideDisabled() throws Exception {
+        MockSearchFunction mockSearch = mockedSearchFunction(Map.of("elastic.co", Map.of("globalRank", 451, "tldRank",23, "tld", "co")));
+        ExactMatchProcessor processor = new ExactMatchProcessor("_tag", mockSearch, "_name", "domain", false, false,
+            List.of(new EnrichSpecification("tld", "tld")));
+
+        Map<String, Object> source = new HashMap<>();
+        source.put("domain", "elastic.co");
+        source.put("tld", null);
```

Comment:
what would the behavior be if "tld" was not set to anything? same as null?

Reply:
No, in that case `tld` would be set by the enrich processor, because that field doesn't exist, so nothing will be overwritten.